creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1031,60 +1024,43 @@ class SnapbackSM {
      */
     let additionalSyncIsRequired
-    if (secondaryCaughtUp) {
+    if (secondaryCaughtUpToPrimary) {
       await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
       additionalSyncIsRequired = false
       this.logDebug(`${logMsgString} || Sync completed in ${monitoringTimeMs}ms`)
 
-      // Secondary progressed but has not caught up yet (was behind by more than max export range)
-    } else if (secondaryProgressed) {
+      // Secondary completed sync but is still behind primary since it was behind by more than max export range
+      // Since syncs are all-or-nothing, if secondary clock has increased at all, we know it successfully completed sync
+    } else if (finalSecondaryClock > initialSecondaryClock) {
       await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
       additionalSyncIsRequired = true
-      this.log(`${logMsgString} || Secondary progressed from clock ${initialSecondaryClock} to ${finalSecondaryClock} but hasn't caught up. Enqueuing additional syncRequest.`)
+      this.log(`${logMsgString} || Secondary successfully synced from clock ${initialSecondaryClock} to ${finalSecondaryClock} but hasn't caught up to Primary. Enqueuing additional syncRequest.`)
 
-      // (1) secondary did not catch up AND (2) secondary did not progress
+      // (1) secondary did not catch up to primary AND (2) secondary did not complete sync
     } else {
       await SecondarySyncHealthTracker.recordFailure(secondaryUrl, userWallet, syncType)
 
-      // only enqueue an additional sync if we have not already met threshold of daily failures
-      /**
-       * todayCount = secondarysynchealthtracker.getToday's syncFailure count (secondary, wallet, syncType, 'Failure')
-       * if (todayCount > threshold) {
-       *    logMsg secondary met daily failure count threshold
-       * } else {
-       *    additionalSyncIsRequired = true
-       * }
-       */
-      const secondaryUserSyncFailureCountTodayResp = SecondarySyncHealthTracker.getSyncRequestOutcomeMetrics({
-        secondary: secondaryUrl,
-        wallet: userWallet,
-        syncType,
-        outcome: SecondarySyncHealthTracker.Outcomes.FAILURE
-        /* date automatically defaults to today */
-      })
-      // Above resp is a map with a single key-value pair; need to parse out value & handle case where no key-value pair exists
-      // TODO what if key-val pair doesn't exist, or if multiple exist...
-      const secondaryUserSyncFailureCountToday = Object.entries(secondaryUserSyncFailureCountTodayResp)[0][1]
-      if (secondaryUserSyncFailureCountToday > this.SecondaryUserSyncDailyFailureCountThreshold) {
+      const secondaryUserSyncFailureCountToday = await SecondarySyncHealthTracker.getSecondaryUserSyncFailureCountToday(
+        secondaryUrl, userWallet, syncType
+      )
+
+      const SecondaryUserSyncDailyFailureCountThreshold = this.nodeConfig.get('secondaryUserSyncDailyFailureCountThreshold')
```

Comment:
since this value does not change, should we declare this variable in the global scope or constructor?

Reply:
fair, done